src/server/components/dashboard/listsItems/itemsRouter.ts

```diff
@@ -38,18 +50,31 @@ listRouter.param("listId", async (req, res, next, listId) => {
   }
 });
 
-listRouter.all("/:listId", listsEditController);
-listRouter.all("/:listId/*", redirectIfUnauthorised);
+listRouter.all("/:listId*", validateAccessToList);
+
+listRouter.get("/:listId", listsEditController);
+listRouter.post("/:listId", listsEditPostController);
```

Comment:
if there is a `if(req.method === "method")` inside a route handler, it can usually be split into get and post.